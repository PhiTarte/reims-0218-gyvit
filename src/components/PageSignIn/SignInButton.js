import React from "react"
import { Button } from "reactstrap"

const SignInButton = () => (
  <div>
    <Button
      style={{
        width: "192px",
        height: "54px",
        borderRadius: "100px",
        backgroundImage: "linear-gradient(to left, #57aad1, #7a57d1)"
      }}
      color="info"
    >
      Sign in
    </Button>
  </div>
)

export default SignInButton
