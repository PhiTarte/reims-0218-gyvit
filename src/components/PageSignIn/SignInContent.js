import React from "react"
import { Button, Row, Container, Col } from "reactstrap"

const SignInContent = () => (
  <div
    class="d-none d-xl-block d-lg-none"
    style={{
      width: "max-width",
      height: "100%",
      backgroundImage:
        "url('https://www.onecomputer.fr/wp-content/uploads/2016/01/blurred-background-81.jpg')",
      position: "relative"
    }}
  >
    <Container fluid>
      <Row>
        <Col xs="9">
          <p
            style={{
              marginTop: "49px",
              fontSize: "16px",
              fontWeight: "normal",
              fontStyle: "normal",
              letterSpacing: "normal",
              textAlign: "left",
              color: "#3b425d",
              position: "absolute",
              marginLeft: "70%"
            }}
          >
            Don't have an account ?
          </p>
        </Col>
        <Col xs="3">
          <Button
            outline
            style={{
              width: "156px",
              height: "54px",
              borderRadius: "100px",
              border: "solid 1px #7a57d1",
              color: "#7a57d1",
              marginTop: "30px"
            }}
          >
            Get Started
          </Button>
        </Col>
      </Row>
    </Container>
  </div>
)

export default SignInContent
