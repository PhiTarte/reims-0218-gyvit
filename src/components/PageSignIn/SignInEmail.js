//TODO: Rename this SigninForm
// Add the signin button
// Make it a presentational (function) componenent
// 3 function props onEmailChange, onPasswordChange, onSubmit
// 2 value props email and password
// Create a SigninContainer that will have a this.state.email and this.state.password value
// and functions handleEmailChange and handlePasswordChange that will do setState
// submitLogin that will do login fetch then stores token and do retrieveMe fetch then dispatch retrieveMeSuccess

import React from "react"
import { Form, FormGroup, Label, Input, NavLink } from "reactstrap"

export default class SignInEmail extends React.Component {
  render() {
    return (
      <Form
        style={{
          marginBottom: "40px",
          marginTop: "60px"
        }}
      >
        <FormGroup>
          <Label for="email">Email
          
           </Label>
          <Input type="email" name="email" id="IdEmail" placeholder="Email" style={{width:"80%"}}/>
        </FormGroup>
        <FormGroup>
          <Label style={{ marginTop: "30px" }} for="password">
            Password
          </Label>
          <NavLink
            style={{
              display: "inline",
              fontSize: "14px",
              fontWeight: "normal",
              fontStyle: "normal",
              fontStretch: "normal",
              lineHeight: "normal",
              letterSpacing: "normal",
              textAlign: "right",
              color: "#7a57d1",
              paddingLeft: "10px",
              width:"50%"

            }}
            href="#"
          >
            Forgot password?
          </NavLink>

          <Input
           style={{width:"80%"}}
            type="password"
            name="password"
            id="IdPassword"
            placeholder="Password"
          />
        </FormGroup>
      </Form>
    )
  }
}
