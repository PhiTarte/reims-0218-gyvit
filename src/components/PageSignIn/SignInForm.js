//TODO: Rename this SigninForm done
// Add the signin button  done
// Make it a presentational (function) componenent
// 3 function props onMailChange, onPasswordChange, onSubmit
// 2 value props mail and password
// Create a SigninContainer that will have a this.state.mail and this.state.password value
// and functions handlemailChange and handlePasswordChange that will do setState
// submitLogin that will do login fetch then stores token and do retrieveMe fetch then dispatch retrieveMeSuccess

import React from "react"
import { Form, FormGroup, Label, Input, NavLink, Button } from "reactstrap"

const SignInForm = ({
  mail,
  password,
  onMailChange,
  onPasswordChange,
  onSubmit
}) => (
  <div>
    <Form
      style={{
        marginBottom: "40px",
        marginTop: "60px"
      }}
    >
      <FormGroup>
        <Label for="mail">Email </Label>
        <Input
          style={{ width: "80%" }}
          value={mail}
          onChange={onMailChange}
          type="email"
          name="mail"
          id="Idmail"
          placeholder="Mail"
        />
      </FormGroup>

      <FormGroup>
        <Label style={{ marginTop: "30px" }} for="password">
          Password
        </Label>
        <NavLink
          style={{
            display: "inline",
            fontSize: "14px",
            fontWeight: "normal",
            fontStyle: "normal",
            fontStretch: "normal",
            lineHeight: "normal",
            letterSpacing: "normal",
            textAlign: "right",
            color: "#7a57d1",
            paddingLeft: "10px",
            width: "50%"
          }}
          href="#"
        >
          Forgot password?
        </NavLink>
        <Input
          style={{ width: "80%" }}
          value={password}
          onChange={onPasswordChange}
          type="password"
          name="password"
          id="IdPassword"
          placeholder="Password"
        />
      </FormGroup>
      <Button
        type="button"
        onClick={onSubmit}
        style={{
          width: "192px",
          height: "54px",
          marginTop: "40px",
          borderRadius: "100px",
          backgroundImage: "linear-gradient(to left, #57aad1, #7a57d1)"
        }}
        //color="info"
      >
        Sign in
      </Button>
    </Form>
  </div>
)

export default SignInForm
