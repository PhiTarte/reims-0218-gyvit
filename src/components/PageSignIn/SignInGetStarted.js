//TODO CREATE SIGN IN TITLE
import React from "react"
import { NavLink } from "reactstrap"

const SignInGetStarted = () => (
  <div style={{ marginTop: "2em" }}>
    <p>
      Don't have an account ?
      <NavLink
        href="#"
        style={{
          display: "inline"
        }}
      >
        Get started
      </NavLink>
    </p>
  </div>
)

export default SignInGetStarted
