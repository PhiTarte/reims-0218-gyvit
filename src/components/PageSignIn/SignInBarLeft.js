import React from "react"
import { Container } from "reactstrap"
import SignInTitle from "./SignInTitle"
import SignInForm from "./SignInForm"
import SignInGetStarted from "./SignInGetStarted"

const PageSignIn = () => (
  <Container
    style={{
      paddingTop: "290px",
      marginLeft: "10%",
      backgroundColor: "#ffffff",
      marginRight: "50px"
    }}
  >
    <SignInTitle />
    <SignInForm />
  
    <SignInGetStarted />
  </Container>
)

export default PageSignIn
