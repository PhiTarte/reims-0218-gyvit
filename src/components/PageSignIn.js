import React from "react"
import { Row, Col, Container } from "reactstrap"
import SignInBarLeft from "../components/SignInBarLeft"
import SignInContent from "./SignInContent"

const PageSignIn = () => (
  <Container fluid>
    <Row>
      <Col md="4">
        <SignInBarLeft />
      </Col>
      <Col md="8">
        <SignInContent />
      </Col>
    </Row>
  </Container>
)

export default PageSignIn
