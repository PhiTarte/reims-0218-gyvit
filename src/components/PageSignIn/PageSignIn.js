import React from "react"
import { Row, Col, Container } from "reactstrap"
import SignInBarLeft from "./SignInBarLeft"
import SignInContent from "./SignInContent"
import ModalErrorContainer from "../../containers/ModalErrorContainer"
const PageSignIn = () => (
  <Container fluid>
    <ModalErrorContainer />
    <Row>
      <Col md="5" lg="5" sm="5">
        <SignInBarLeft />
      </Col>
      <Col md="7" lg="7" sm="7">
        <SignInContent />
      </Col>
    </Row>
  </Container>
)

export default PageSignIn
