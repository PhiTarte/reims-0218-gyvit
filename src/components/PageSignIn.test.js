//TODO: test:  navbar top with search bar, notification icon and profile.

import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"
import PageSignIn from "./PageSignIn"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<PageSignIn />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it("renders correctly", () => {
  const tree = renderer.create(<PageSignIn />).toJSON()
  expect(tree).toMatchSnapshot()
})
