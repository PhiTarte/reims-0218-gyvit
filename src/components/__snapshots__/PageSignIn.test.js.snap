// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
  style={
    Object {
      "background-color": "#ffffff",
      "box-shadow": "0 3px 14px 0 rgba(224, 227, 240, 0.6)",
      "margin-left": "60px",
      "margin-top": "290px",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div
        className="ml-3 mr-3"
      >
        <div>
          <h1
            style={
              Object {
                "color": "#3b425d",
                "font-size": "24px",
                "font-stretch": "normal",
                "font-style": "normal",
                "font-weight": "normal",
                "height": "24px",
                "letter-spacing": "normal",
                "line-height": "normal",
                "text-align": "left",
              }
            }
          >
            Sign it to Gyvit
          </h1>
          <p
            style={
              Object {
                "color": "#9da5c9",
                "font-size": "18px",
                "font-stretch": "normal",
                "font-style": "normal",
                "font-weight": "normal",
                "height": "18px",
                "letter-spacing": "normal",
                "line-height": "normal",
                "margin-top": "15px",
                "text-align": "left",
                "width": "200px",
              }
            }
          >
            Enter your details below
             
          </p>
        </div>
        <form
          className=""
          style={
            Object {
              "margin-bottom": "40px",
              "margin-top": "60px",
            }
          }
        >
          <div
            className="form-group"
          >
            <label
              className=""
              htmlFor="email"
            >
              Email 
            </label>
            <input
              className="form-control"
              id="IdEmail"
              name="email"
              placeholder="Email"
              type="email"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className=""
              htmlFor="password"
              style={
                Object {
                  "margin-top": "30px",
                }
              }
            >
              Password
            </label>
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "color": "#7a57d1",
                  "display": "inline",
                  "font-size": "14px",
                  "font-stretch": "normal",
                  "font-style": "normal",
                  "font-weight": "normal",
                  "letter-spacing": "normal",
                  "line-height": "normal",
                  "padding-left": "100px",
                  "text-align": "right",
                }
              }
            >
              Forgot password?
            </a>
            <input
              className="form-control"
              id="IdPassword"
              name="password"
              placeholder="Password"
              type="password"
            />
          </div>
        </form>
        <div>
          <button
            className="btn btn-info"
            onClick={[Function]}
            style={
              Object {
                "background-image": "linear-gradient(to left, #57aad1, #7a57d1)",
                "border-radius": "100px",
                "height": "54px",
                "width": "192px",
              }
            }
            type={undefined}
          >
            Sign in
          </button>
        </div>
        <div
          style={
            Object {
              "margin-top": "120px",
            }
          }
        >
          <p>
            Don't have an account ?
            <a
              className="nav-link"
              href="#"
              onClick={[Function]}
              style={
                Object {
                  "display": "inline",
                }
              }
            >
              Get started
            </a>
          </p>
        </div>
        <div
          style={
            Object {
              "background": "green",
              "height": "1067px",
              "width": "980px",
            }
          }
        >
          <button
            className="btn btn-info"
            onClick={[Function]}
            style={
              Object {
                "background-image": "linear-gradient(to left, #57aad1, #7a57d1)",
                "border-radius": "100px",
                "height": "54px",
                "width": "192px",
              }
            }
            type={undefined}
          >
            Sign in
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
